cli/src/main/java/com/liferay/blade/cli/BladeCLI.java

```diff
@@ -112,15 +112,37 @@ public BaseArgs getArgs() {
 	}
 
 	public BladeSettings getBladeSettings() throws IOException {
-		final File settingsFile;
+		File settingsBaseDir;
 
 		if (WorkspaceUtil.isWorkspace(this)) {
-			File workspaceDir = WorkspaceUtil.getWorkspaceDir(this);
-
-			settingsFile = new File(workspaceDir, ".blade/settings.properties");
+			settingsBaseDir = WorkspaceUtil.getWorkspaceDir(this);
 		}
 		else {
-			settingsFile = new File(_USER_HOME_DIR, ".blade/settings.properties");
+			settingsBaseDir = _USER_HOME_DIR;
+		}
+
+		File settingsFile = new File(settingsBaseDir, BladeSettings.BLADE_SETTINGS_OLD_STRING);
+
+		if (settingsFile.exists()) {
+			String name = settingsFile.getName();
+
+			if ("settings.properties".equals(name)) {
+				Path settingsPath = settingsFile.toPath();
+
+				Path settingsParentPath = settingsPath.getParent();
+
+				if (settingsParentPath.endsWith(".blade")) {
+					Path settingsParentParentPath = settingsParentPath.getParent();
+
+					Path newSettingsPath = settingsParentParentPath.resolve(BladeSettings.BLADE_SETTINGS_NEW_STRING);
+
+					Files.move(settingsPath, newSettingsPath);
```

Comment:
This method name is `get` we should never make mutable changes in a 'getter'.  Lets refactor the "migration" to a new private method in BladeCLI.

Reply:
Good idea